src/Featurizers/Components/StatisticsMetricsEstimator.h

```diff
@@ -0,0 +1,391 @@
+    // |
+    // ----------------------------------------------------------------------
+    BasicStatisticalAnnotationData(T min, T max, std::uint64_t count);
+    ~BasicStatisticalAnnotationData(void) = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(BasicStatisticalAnnotationData);
+};
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         StandardStatisticalAnnotaionData
+///  \brief         An annotation class which contains the sum and average
+///                 for an input column. Template T can only be numerical types
+///
+template <typename T>
+class StandardStatisticalAnnotaionData : public BasicStatisticalAnnotationData<T> {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Data
+    // |
+    // ----------------------------------------------------------------------
+    long double   const                            Sum;
+    std::double_t const                            Average;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    StandardStatisticalAnnotaionData(long double sum, std::double_t average, T min, T max, std::uint64_t count);
+    ~StandardStatisticalAnnotaionData(void) = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(StandardStatisticalAnnotaionData);
+};
+
+namespace Details {
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         BasicStatsTrainingOnlyPolicy
+///  \brief         BasicStatsTrainingOnlyPolicy deals with basic stats include
+///                 min, max and count
+///
+template <typename T>
+class BasicStatsTrainingOnlyPolicy {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using InputType                         = T;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Data
+    // |
+    // ----------------------------------------------------------------------
+    static constexpr char const * const     NameValue = StatisticalMetricsEstimatorName;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    BasicStatsTrainingOnlyPolicy(void);
+
+    void fit(InputType const &input);
+    BasicStatisticalAnnotationData<T> complete_training(void);
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Data
+    // |
+    // ----------------------------------------------------------------------
+    InputType                                 _min;

```

Comment:
These shouldn't be public. Consider making them protected so StandardStatsTrainingOnlyPolicy has access to them.